onsite/global_flr.py

```
@@ -0,0 +1,521 @@
+            )
+
+        plt.tight_layout()
+        plt.savefig(output_path, dpi=150, bbox_inches='tight')
+        plt.close()
+
+        logger.info(f"Cumulative FLR curve saved to: {output_path}")
+
+    def generate_summary_report(
+        self,
+        output_path: str,
+        threshold: float = 0.01
```

Ruff (0.14.13)

Unused method argument: `threshold`

(ARG002)